diff --git a/openembedded-core/meta/conf/bitbake.conf b/openembedded-core/meta/conf/bitbake.conf
index 5e98d4511f..8ec32a8a4f 100644
--- a/openembedded-core/meta/conf/bitbake.conf
+++ b/openembedded-core/meta/conf/bitbake.conf
@@ -533,8 +533,8 @@ BUILDSDK_CPPFLAGS = ""
 export CPPFLAGS = "${TARGET_CPPFLAGS}"
 export TARGET_CPPFLAGS = ""
 
-export BUILD_CFLAGS = "${BUILD_CPPFLAGS} ${BUILD_OPTIMIZATION}"
-BUILDSDK_CFLAGS = "${BUILDSDK_CPPFLAGS} ${BUILD_OPTIMIZATION}"
+export BUILD_CFLAGS = "${BUILD_CPPFLAGS} ${BUILD_OPTIMIZATION} -Wno-error -Wno-error=format-nonliteral"
+BUILDSDK_CFLAGS = "${BUILDSDK_CPPFLAGS} ${BUILD_OPTIMIZATION} -Wno-error -Wno-error=format-nonliteral"
 export CFLAGS = "${TARGET_CFLAGS}"
 export TARGET_CFLAGS = "${TARGET_CPPFLAGS} ${SELECTED_OPTIMIZATION}"
 
