--- ./openembedded-core/meta/lib/oe/package_manager.py
+++ ./openembedded-core/meta/lib/oe/package_manager.py
@@ -1348,6 +1348,9 @@ class OpkgPM(OpkgDpkgPM):
                                               (cmd, e.returncode, e.output.decode("utf-8")))
 
     def remove(self, pkgs, with_dependencies=True):
+        #temporary OE-core rootfs fix
+        if len(pkgs) == 0:
+            return
         if with_dependencies:
             cmd = "%s %s --force-remove --force-removal-of-dependent-packages remove %s" % \
                 (self.opkg_cmd, self.opkg_args, ' '.join(pkgs))
